client/reader/main.cpp

```diff
@@ -0,0 +1,33 @@
+#include "reader.h"
+#include <iostream>
+
+#define MAXSIZE 100
+
+int main(int argc, char *argv[]) {
+
+  if (argc != 3) {
+    std::cout << argv[0] << " ip port" << std::endl;
+    return 1;
+  }
+
+  Reader *reader = Reader::instance();
```

Comment:
Reader should be a regular class, not a singleton.
However, even if you decided to make some class a singleton, you should not assign the instance of the class to a variable. That leads to problems in future. Like potentially another dev can try to free it.